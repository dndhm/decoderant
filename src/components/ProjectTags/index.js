import React from 'react';
import propTypes from 'prop-types';
import injectStaticSheet from 'inject-static-sheet';

import colors, { MID_GREY } from '../../constants/colors';

const styles = {
	list: {
		margin: '0 auto 10px',
		'text-align': 'center',
	},
	listItem: {
		color: colors[MID_GREY],
		display: 'inline',
		'font-size': '11px',
		'margin-right': '3px',
		'padding-left': '10px',
		'text-transform': 'uppercase',
	},
	'listItem:nth-of-type(1)': {
		'padding-left': 0,
	}
};

const ProjectTags = ({ classes, tags }) => (
	<ul className={classes.list}>
		{tags.map((tag, index) => (
			<li
				className={classes.listItem}
				key={index}
			>
				{tag}
			</li>
		))}
	</ul>
);

export const tags = [
	'Backbone',
	'CSS',
	'JS',
	'Phonegap',
	'PHP',
	'React',
	'Redux',
	'Responsive',
	'Wordpress',
];

ProjectTags.propTypes = {
	classes: propTypes.object,
	tags: propTypes.arrayOf(
		propTypes.oneOf(tags)
	).isRequired,
};

export default injectStaticSheet(styles, 'ProjectTags')(ProjectTags);
