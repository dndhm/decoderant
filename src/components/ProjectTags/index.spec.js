import React from 'react';
import { create } from 'react-test-renderer';

import ProjectTags, { tags } from './';

describe('ProjectTags', () => {
	tags.forEach(tag => {
		test(`should render correctly for given tag: ${tag}`, () => {
			const tree = create(
				<ProjectTags tags={[tag]} />
			)
				.toJSON();
			expect(tree).toMatchSnapshot();
		});

		test('should render correctly with multiple tags', () => {
			const tree = create(
				<ProjectTags tags={tags} />
			)
				.toJSON();
			expect(tree).toMatchSnapshot();
		});
	});
});
