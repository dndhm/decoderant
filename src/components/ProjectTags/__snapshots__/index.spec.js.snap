// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectTags should render correctly for given tag: Backbone 1`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    Backbone
  </li>
</ul>
`;

exports[`ProjectTags should render correctly for given tag: CSS 1`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    CSS
  </li>
</ul>
`;

exports[`ProjectTags should render correctly for given tag: JS 1`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    JS
  </li>
</ul>
`;

exports[`ProjectTags should render correctly for given tag: PHP 1`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    PHP
  </li>
</ul>
`;

exports[`ProjectTags should render correctly for given tag: Phonegap 1`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    Phonegap
  </li>
</ul>
`;

exports[`ProjectTags should render correctly for given tag: React 1`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    React
  </li>
</ul>
`;

exports[`ProjectTags should render correctly for given tag: Redux 1`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    Redux
  </li>
</ul>
`;

exports[`ProjectTags should render correctly for given tag: Responsive 1`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    Responsive
  </li>
</ul>
`;

exports[`ProjectTags should render correctly for given tag: Wordpress 1`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    Wordpress
  </li>
</ul>
`;

exports[`ProjectTags should render correctly with multiple tags 1`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    Backbone
  </li>
  <li
    className="ProjectTags--listItem"
  >
    CSS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    JS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Phonegap
  </li>
  <li
    className="ProjectTags--listItem"
  >
    PHP
  </li>
  <li
    className="ProjectTags--listItem"
  >
    React
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Redux
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Responsive
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Wordpress
  </li>
</ul>
`;

exports[`ProjectTags should render correctly with multiple tags 2`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    Backbone
  </li>
  <li
    className="ProjectTags--listItem"
  >
    CSS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    JS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Phonegap
  </li>
  <li
    className="ProjectTags--listItem"
  >
    PHP
  </li>
  <li
    className="ProjectTags--listItem"
  >
    React
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Redux
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Responsive
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Wordpress
  </li>
</ul>
`;

exports[`ProjectTags should render correctly with multiple tags 3`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    Backbone
  </li>
  <li
    className="ProjectTags--listItem"
  >
    CSS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    JS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Phonegap
  </li>
  <li
    className="ProjectTags--listItem"
  >
    PHP
  </li>
  <li
    className="ProjectTags--listItem"
  >
    React
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Redux
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Responsive
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Wordpress
  </li>
</ul>
`;

exports[`ProjectTags should render correctly with multiple tags 4`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    Backbone
  </li>
  <li
    className="ProjectTags--listItem"
  >
    CSS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    JS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Phonegap
  </li>
  <li
    className="ProjectTags--listItem"
  >
    PHP
  </li>
  <li
    className="ProjectTags--listItem"
  >
    React
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Redux
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Responsive
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Wordpress
  </li>
</ul>
`;

exports[`ProjectTags should render correctly with multiple tags 5`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    Backbone
  </li>
  <li
    className="ProjectTags--listItem"
  >
    CSS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    JS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Phonegap
  </li>
  <li
    className="ProjectTags--listItem"
  >
    PHP
  </li>
  <li
    className="ProjectTags--listItem"
  >
    React
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Redux
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Responsive
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Wordpress
  </li>
</ul>
`;

exports[`ProjectTags should render correctly with multiple tags 6`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    Backbone
  </li>
  <li
    className="ProjectTags--listItem"
  >
    CSS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    JS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Phonegap
  </li>
  <li
    className="ProjectTags--listItem"
  >
    PHP
  </li>
  <li
    className="ProjectTags--listItem"
  >
    React
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Redux
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Responsive
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Wordpress
  </li>
</ul>
`;

exports[`ProjectTags should render correctly with multiple tags 7`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    Backbone
  </li>
  <li
    className="ProjectTags--listItem"
  >
    CSS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    JS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Phonegap
  </li>
  <li
    className="ProjectTags--listItem"
  >
    PHP
  </li>
  <li
    className="ProjectTags--listItem"
  >
    React
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Redux
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Responsive
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Wordpress
  </li>
</ul>
`;

exports[`ProjectTags should render correctly with multiple tags 8`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    Backbone
  </li>
  <li
    className="ProjectTags--listItem"
  >
    CSS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    JS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Phonegap
  </li>
  <li
    className="ProjectTags--listItem"
  >
    PHP
  </li>
  <li
    className="ProjectTags--listItem"
  >
    React
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Redux
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Responsive
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Wordpress
  </li>
</ul>
`;

exports[`ProjectTags should render correctly with multiple tags 9`] = `
<ul
  className="ProjectTags--list"
>
  <li
    className="ProjectTags--listItem"
  >
    Backbone
  </li>
  <li
    className="ProjectTags--listItem"
  >
    CSS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    JS
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Phonegap
  </li>
  <li
    className="ProjectTags--listItem"
  >
    PHP
  </li>
  <li
    className="ProjectTags--listItem"
  >
    React
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Redux
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Responsive
  </li>
  <li
    className="ProjectTags--listItem"
  >
    Wordpress
  </li>
</ul>
`;
